### PR TITLE
Remove `new Date` fallback in `toDate` (closes #743)

### LINKED ISSUE #743: Remove `new Date` fallback in `toDate`
This fallback leads to all functions working differently in different browsers (sinse `toDate` is used to clean the date arguments in all functions). For example, `toDate('07-May-2018')` returns `Invalid Date` in Firefox, but correctly parsed date in Chrome. See also #742

### PR DESCRIPTION
**Pull request by leshakoss**
See #743

### CODE REVIEW

File: src/toDate/test.js [hunk 1]
```diff
@@ -259,18 +260,54 @@ describe('toDate', function () {
...
 
-  it('throws `RangeError` if `options.additionalDigits` is not convertable to 0, 1, 2 or undefined`', function () {
-    // $ExpectedMistake
-    var block = toDate.bind(null, '+12340702', {additionalDigits: 3})
-    assert.throws(block, RangeError)
+  describe('argument conversion', function () {
+    it('implicitly converts instance of Number into a number', function () {
+      // eslint-disable-next-line no-new-wrappers
+      var timestamp = new Number(new Date(2016, 0, 1, 23, 30, 45, 123).getTime())
+      // $ExpectedMistake
```
Comment (kossnocorp) [inline src/toDate/test.js]: Hm, why Flow does complaints here?

File: src/toDate/test.js [hunk 2]
```diff
@@ -259,18 +260,54 @@ describe('toDate', function () {
...
+      var result = toDate(timestamp)
+      assert.deepEqual(result, new Date(2016, 0, 1, 23, 30, 45, 123))
+    })
+
+    it('implicitly converts instance of String into a string', function () {
+      // eslint-disable-next-line no-new-wrappers
+      var dateString = new String('2014-02-11')
+      // $ExpectedMistake
```
Comment (kossnocorp) [inline src/toDate/test.js]: Same question here

File: src/toDate/test.js (hunk 1 above)
Comment (leshakoss) [reply src/toDate/test.js]: Oh, it is probably because the argument type is `*` right now. Should we make it `Date|String|Number`?

Comment (kossnocorp) [reply src/toDate/test.js]: Let's do it

**Pull request merged**